cli/deployment/flags.go

```diff
@@ -0,0 +1,447 @@
+			EnvVar: "CODER_TLS_KEY_FILE",
+			Description: `Policy the server will follow for TLS Client Authentication. ` +
+				`Accepted values are "none", "request", "require-any", "verify-if-given", or "require-and-verify"`,
+			Default: "request",
+		},
+		TLSKeyFiles: codersdk.StringArrayFlag{
+			Name:        "TLS Key Files",
+			Flag:        "tls-key-file",
+			EnvVar:      "CODER_TLS_KEY_FILE",
+			Description: "Paths to the private keys for each of the certificates. It requires a PEM-encoded file",
+			Default:     []string{},
+		},
+		TLSMinVersion: codersdk.StringFlag{
+			Name:        "TLS Min Version",
+			Flag:        "tls-min-version",
+			EnvVar:      "CODER_TLS_MIN_VERSION",
+			Description: `Minimum supported version of TLS. Accepted values are "tls10", "tls11", "tls12" or "tls13"`,
+			Default:     "tls12",
+		},
+		TraceEnable: codersdk.BoolFlag{
+			Name:        "Trace Enabled",
+			Flag:        "trace",
+			EnvVar:      "CODER_TRACE",
+			Description: "Whether application tracing data is collected.",
+		},
+		SecureAuthCookie: codersdk.BoolFlag{
+			Name:        "Secure Auth Cookie",
+			Flag:        "secure-auth-cookie",
+			EnvVar:      "CODER_SECURE_AUTH_COOKIE",
+			Description: "Controls if the 'Secure' property is set on browser session cookies",
+		},
+		SSHKeygenAlgorithm: codersdk.StringFlag{
+			Name:   "SSH Keygen Algorithm",
+			Flag:   "ssh-keygen-algorithm",
+			EnvVar: "CODER_SSH_KEYGEN_ALGORITHM",
+			Description: "The algorithm to use for generating ssh keys. " +
+				`Accepted values are "ed25519", "ecdsa", or "rsa4096"`,
+			Default: "ed25519",
+		},
+		AutoImportTemplates: codersdk.StringArrayFlag{
+			Name:        "Auto Import Templates",
+			Flag:        "auto-import-template",
+			EnvVar:      "CODER_TEMPLATE_AUTOIMPORT",
+			Description: "Templates to auto-import. Available auto-importable templates are: kubernetes",
+			Default:     []string{},
+		},
+		MetricsCacheRefreshInterval: codersdk.DurationFlag{
+			Name:        "Metrics Cache Refresh Interval",
+			Flag:        "metrics-cache-refresh-interval",
+			EnvVar:      "CODER_METRICS_CACHE_REFRESH_INTERVAL",
+			Description: "How frequently metrics are refreshed",
+			Default:     time.Hour,
+		},
+		AgentStatRefreshInterval: codersdk.DurationFlag{
+			Name:        "Agent Stats Refresh Interval",
+			Flag:        "agent-stats-refresh-interval",
+			EnvVar:      "CODER_AGENT_STATS_REFRESH_INTERVAL",
+			Description: "How frequently agent stats are recorded",
+			Default:     10 * time.Minute,
+		},
+		Verbose: codersdk.BoolFlag{
+			Name:        "Verbose Logging",
+			Flag:        "verbose",
+			EnvVar:      "CODER_VERBOSE",
+			Shorthand:   "v",
+			Description: "Enables verbose logging.",
+		},
+		AuditLogging: codersdk.BoolFlag{
+			Name:        "Audit Logging",
+			Flag:        "audit-logging",
+			EnvVar:      "CODER_AUDIT_LOGGING",
+			Description: "Specifies whether audit logging is enabled.",
+			Default:     true,
+		},
```

Comment:
It's a bit awkward to have these attached to the AGPL version of Coder and certainly muddies the implementation difference a bit. 

I like the types that are generated though, and I suppose that wouldn't be possible to get without doing something like this.

Did any other ideas of how to split this up come to mind?

Reply:
The other options here I considered were:
- make a `codersdk.EnterpriseDeploymentFlags` that embeds / copies the agpl struct but we can't return different structs for the same route and making 2 routes means you must know the license status first which is less ideal.
- Make an array / map of flags - you lose the ability to click through and see usages of the field directly which I valued for discoverability.

I asked @deansheather and @bpmct who looked over my shoulder working on this and we felt this was a good middle ground.

Reply:
Seems fine to me